src/Tasks/Microsoft.NET.Build.Tasks/LockFileExtensions.cs

```diff
@@ -76,6 +83,38 @@ public static LockFileTargetLibrary GetLibrary(this LockFileTarget lockFileTarge
 
             return projectDeps;
         }
+        
+        public static HashSet<string> GetIntersetction(
```

Comment:
This method doesn't really seem to belong in LockFileExtensions, since it is not an extension method.

My thought would be to make it a private method next to where it is called, and if it needs to be shared somewhere, it can be moved to an appropriate place later.